XamlControlsGallery/NewControlsPage.xaml.cs

```diff
@@ -26,35 +27,74 @@ public NewControlsPage()
+
+            Items = ControlInfoDataSource.Instance.Groups.SelectMany(g => g.Items.Where(i => i.BadgeString != null)).OrderBy(i => i.Title).ToList();
+            itemsCVS.Source = FormatData();
         }
 
-        private IEnumerable<ControlInfoDataItem> _previewitems;
+        private ObservableCollection<GroupInfoList> FormatData()
+        {
+            var query = from item in this.Items
+                        group item by item.BadgeString into g
+                        orderby g.Key
+                        select new GroupInfoList(g) { Key = g.Key };
+
+            ObservableCollection<GroupInfoList> groupList = new ObservableCollection<GroupInfoList>(query);
+
+            //Move Preview samples to the back of the list
+            var previewGroup = groupList.ElementAt(1);
+            if (previewGroup?.Key.ToString() == "Preview")
+            {
+                groupList.RemoveAt(1);
+                groupList.Insert(groupList.Count, previewGroup);
+            }
 
-        public IEnumerable<ControlInfoDataItem> PreviewItems
+            foreach (var item in groupList)
+            {
+                switch (item.Key.ToString())
+                {
+                    case "New":
+                        item.Title = "Recently Added Samples";
+                        break;
+                    case "Updated":
+                        item.Title = "Recently Updated Samples";
+                        break;
+                    case "Preview":
+                        item.Title = "Preview Samples";
+                        break;
+                }
+            }
+
+            return groupList;
+        }
+
+        protected override bool GetIsNarrowLayoutState()
         {
-            get { return _previewitems; }
-            set { SetProperty(ref _previewitems, value); }
+            return LayoutVisualStates.CurrentState == NarrowLayout;
         }
+    }
 
-        protected override void OnNavigatedTo(NavigationEventArgs e)
+    public class GroupInfoList : List<object>
+    {
+        public GroupInfoList(IEnumerable<object> items) : base(items)
         {
-            var menuItem = NavigationRootPage.Current.NavigationView.MenuItems.Cast<Microsoft.UI.Xaml.Controls.NavigationViewItem>().First();
-            menuItem.IsSelected = true;
-            NavigationRootPage.Current.NavigationView.Header = menuItem.Content;
-            Items = ControlInfoDataSource.Instance.Groups.SelectMany(g => g.Items.Where(i => i.IsNew)).OrderBy(i => i.Title).ToList();
-            UpdatedItems = ControlInfoDataSource.Instance.Groups.SelectMany(g => g.Items.Where(i => i.IsUpdated)).OrderBy(i => i.Title).ToList();
-            PreviewItems = ControlInfoDataSource.Instance.Groups.SelectMany(g => g.Items.Where(i => i.IsPreview)).OrderBy(i => i.Title).ToList();
         }
+        public object Key { get; set; }
 
-        protected override bool GetIsNarrowLayoutState()
+        private string _title;
```

Comment:
This can be replaced with an auto property, but it does not make much of a difference in the end. 😅